Penumbra/UI/ModsTab/ModPanelTagsFeatureTab.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using ImGuiNET;
+using OtterGui.Raii;
+using OtterGui;
+using OtterGui.Widgets;
+using Penumbra.Mods.Manager;
+using System.Numerics;
+using System.Collections.Generic;
+using Dalamud.Logging;
+using System.Linq;
+using Penumbra.UI.Classes;
+
+namespace Penumbra.UI.ModsTab;
+
+public class ModPanelTagsFeatureTab : ITab
+{
+    
+    private readonly ModFileSystemSelector _selector;
+    private readonly TutorialService _tutorial;
+    private readonly ModManager _modManager;
+    private readonly TagButtons _localTags = new();
+
+    //Contains the tags
+    private List<List<string>> _tagsList = new();
+    //private Dictionary<string, bool> _tagDict = new();
+    //private List<string> _selectedTags = new();
+
+    public ModPanelTagsFeatureTab(ModFileSystemSelector selector, TutorialService tutorial, ModManager modManager)
+    {
+        _selector = selector;
+        _tutorial = tutorial;
+        _modManager = modManager;
+
+        AddTagsToList();
+    }
+
+    public ReadOnlySpan<byte> Label
+        => "Tag Test Feature"u8;
+
+    public void DrawContent()
+    {
+        //Maybe remove this from the description tab?
+        var tagIdx = _localTags.Draw("Local Tags: ",
+            "Custom tags you can set personally that will not be exported to the mod data but only set for you.\n"
+          + "If the mod already contains a local tag in its own tags, the local tag will be ignored.", _selector.Selected!.LocalTags,
+            out var editedTag, false);
+        _tutorial.OpenTutorial(BasicTutorialSteps.Tags);
+        if (tagIdx >= 0)
+            _modManager.DataEditor.ChangeLocalTag(_selector.Selected!, tagIdx, editedTag);
+
+
+        //Adds the buttons, right now the tags are just in a List that contains List<string>, I'd properly move the tags to a json file so its easier to add and remove tags, can also add a function so the users can add tags themselves
```

Comment:
Categories should probably just carry their own names in the list, making it a 
```cs
List<(string Name, List<string> Tags)>
```
instead.
Then you could use a foreach, too. Also better when letting users edit it.